src/pal/src/file/file.cpp

```diff
@@ -1136,14 +1168,17 @@ DeleteFileA(
     ENTRY("DeleteFileA(lpFileName=%p (%s))\n", lpFileName?lpFileName:"NULL", lpFileName?lpFileName:"NULL");
 
     pThread = InternalGetCurrentThread();
-    if (strlen(lpFileName) >= MAX_LONGPATH)
+    length = strlen(lpFileName);
+    if (length >= MAX_LONGPATH)
```

Comment:
The MAX_LONGPATH is an arbitrary constant that we have chosen, it has nothing to do with the OS limits. And to make it even more interesting, not even PATH_MAX defined in limits.h represents the limit of the actual OS or even actual filesystem on an OS. There is a path_conf Linux API that is able to extract the real path limits for a specific filesystem path. That one can even return -1 in case the size is unlimited.